core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -448,6 +449,107 @@ public LogicalPlan visitEval(Eval node, AnalysisContext context) {
     return new LogicalEval(child, expressionsBuilder.build());
   }
 
+  /**
+   * Builds and returns a {@link org.opensearch.sql.planner.logical.LogicalFlatten} corresponding to
+   * the given flatten node, and adds the new fields to the current type environment.
+   *
+   * <p><b>Example</b>
+   *
+   * <p>Input Data:
+   *
+   * <pre>
+   * [
+   *   struct: {
+   *     integer: 0,
+   *     nested_struct: { string: "value" }
+   *   }
+   * ]
+   * </pre>
+   *
+   * Query 1: <code>flatten struct</code>
+   *
+   * <pre>
+   * [
```

Comment:
Change from array outer structure to object
```suggestion
   * {
```

Reply:
Updated!